src/service/iplayerService.ts

```diff
@@ -278,15 +278,15 @@ const iplayerService = {
                 infos = [...infos, ...chunkInfos];
             }
 
-            // const infos = await iplayerService.details(resultPids);
-            return await Promise.all(infos.map((info: IPlayerDetails) => createResult(info.title, info, sizeFactor)));
+            const synonymName = synonym ? (synonym.filenameOverride || synonym.from).replaceAll(/[^a-zA-Z0-9\s.]/g, '').replaceAll(' ', '.') : undefined;
```

Comment:
This is the existing logic for calculating the `synonymName`, I just moved it up so it was calculated once rather than per result.